sklearn/tree/_splitter.pyx

```diff
@@ -817,26 +855,64 @@ cdef inline int node_split_random(
         current_proxy_improvement = criterion.proxy_impurity_improvement()
 
         if current_proxy_improvement > best_proxy_improvement:
+            current_split.n_missing = n_missing
+
+            if n_missing == 0:
+                current_split.missing_go_to_left = n_left > n_right
```

Comment:
Is this required for the case where there are no missing values in the training data but there can be during inference?